src/FlowTransformer.cpp

```diff
@@ -0,0 +1,459 @@
+    //ROS_ERROR_STREAM("Current altitude before use: " << current_altitude_);
+    double distance_to_plane = current_altitude_
+                             * std::sqrt(1.0
+                            + std::pow(std::tan(pitch), 2.0)
+                            + std::pow(std::tan(roll), 2.0));
+
+    //ROS_ERROR("distance_to_plane is %lf\n", distance_to_plane);
+    
+    
+    // Multiplier that converts measurements in pixels to measurements in
+    // meters in the plane parallel to the camera's xy plane and going through
+    // the point Pg, where Pg is intersection between the camera's +z axis and
+    // the ground plane
+    //
+    // Focal length gives distance from camera to image plane in pixels, so
+    // dividing distance to plane by this number gives the multiplier we want
+    // dtp * tan 42 /15 px
+
+    double current_meters_per_px = distance_to_plane * std::tan(flow_transformer_settings.fov/2)
+                                    / (flow_transformer_settings.image_width/2.0);
+
+    //ROS_ERROR_STREAM("current meters per pix: " << current_meters_per_px);
+
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    // These velocities are flipped, but it really should be implemented properly in the 
+    // transform tree. Definitely fix this later.
+    // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+    float estimatedXVel = current_meters_per_px * deltaX / std::cos(pitch)/ dt;
+    float estimatedYVel = current_meters_per_px * deltaY / -std::cos(roll) / dt;
+    //ROS_ERROR("estimatedX, %f dX, %d, estimatedY, %f, dY, %d\n", estimatedXVel, deltaX, estimatedYVel, deltaY);
+
+    // Move the samples taken forward in time
+    for(int i = filter_order_-1; i > -1; i--)
+    {
+        velocity_filtered_[0][i+1] = velocity_filtered_[0][i];
+        velocity_filtered_[1][i+1] = velocity_filtered_[1][i];
+    }
+
+    velocity_filtered_[0][0] = estimatedXVel;
+    velocity_filtered_[1][0] = estimatedYVel;
+
+
+    float filteredXVel = 0;
+    float filteredYVel = 0;
+
+
+    for(int i = 0; i < filter_order_ + 1; i++)
+    {
+        //ROS_ERROR_STREAM("velocity filtered X: " << velocity_filtered_[0][i] << " filter coefs: " << filter_coefs_[i]);
+        //ROS_ERROR_STREAM("velocity filtered Y: " << velocity_filtered_[1][i] << " filter coefs: " << filter_coefs_[i]);
+        filteredXVel += velocity_filtered_[0][i] * filter_coefs_[i];
+        filteredYVel += velocity_filtered_[1][i] * filter_coefs_[i];
+    }
+    //ROS_ERROR_STREAM("Filtered X Vel: " << filteredXVel << " estimated X Vel: " << estimatedXVel);
+    //ROS_ERROR_STREAM("Filtered Y Vel: " << filteredYVel << " estimated Y Vel: " << estimatedYVel);
+
+    //filteredXVel = estimatedXVel;
+    //filteredYVel = estimatedYVel;
+
+    /*double dp;
+    double dr;
+
+    // These two if statements make sure that dp and dr are the shortest change
```

Comment:
Half of this file is commented out. Is the code not needed?

Reply:
I am seeing some questionable things I left behind. I will remove them when I get a chance